Dockerfiles/agent/Dockerfile

```diff
@@ -1,25 +1,53 @@
+FROM debian:stretch-slim as extract
+ARG WITH_JMX
+COPY datadog-agent*_amd64.deb /
+WORKDIR /output
+
+# Extract and cleanup:
+#   - unused systemd unit
+#   - GPL sources for embedded software
+#   - docs and manpages
+#   - static libraries
+#   - jmxfetch on nojmx build
+RUN dpkg -x /datadog-agent*_amd64.deb . \
+ && rm -rf usr etc/init lib \
+    opt/datadog-agent/sources \
+    opt/datadog-agent/embedded/share/doc \
+    opt/datadog-agent/embedded/share/man \
+ && find opt/datadog-agent/ -iname "*.a" -delete \
```

Comment:
we should avoid to load most of these things into the package in the first place, I'd move these 3 lines somewhere into the Omnibus installer (off the top of my head, I'd put these in the finalize software definition, see https://github.com/DataDog/datadog-agent/blob/master/omnibus/config/software/datadog-agent-finalize.rb)

Reply:
I disagree **_slightly_** with this. Not the end-goal, but the fact that we're handling it in this manner. We should fix the definitions at build-time so that the static libs aren't built/installed in the first place (same though definitely less critical and something I could live with for the manpages, documentation, etc). The fact we have the static libraries already available in the path when other libraries/objects are getting built and linked is already "dangerous". And this should definitely be done in omnibus so that we at least get the healthcheck to verify we're not removing anything needed.

Reply:
Even better ^

Reply:
I do agree, although you don't want me messing up with omnibus recipes 😇

Reply:
sounds good - this backlog card should mention that once it's completed we should remove these steps here